app/src/main/java/tech/pauly/findapet/shared/MapWrapper.kt

```diff
@@ -1,35 +1,54 @@
 package tech.pauly.findapet.shared
 
+import android.annotation.SuppressLint
 import android.arch.lifecycle.Lifecycle
 import android.arch.lifecycle.LifecycleObserver
 import android.arch.lifecycle.OnLifecycleEvent
 import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.support.annotation.DrawableRes
 import com.google.android.gms.maps.CameraUpdateFactory
 import com.google.android.gms.maps.GoogleMap
 import com.google.android.gms.maps.MapView
 import com.google.android.gms.maps.OnMapReadyCallback
-import com.google.android.gms.maps.model.LatLng
-import com.google.android.gms.maps.model.LatLngBounds
-import com.google.android.gms.maps.model.MapStyleOptions
-import com.google.android.gms.maps.model.MarkerOptions
+import com.google.android.gms.maps.model.*
 import io.reactivex.Observable
 import io.reactivex.subjects.BehaviorSubject
+import io.reactivex.subjects.PublishSubject
 import tech.pauly.findapet.R
 import tech.pauly.findapet.dependencyinjection.ForApplication
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-open class MapWrapper @Inject constructor(@ForApplication val context: Context) : OnMapReadyCallback, LifecycleObserver {
+open class MapWrapper @Inject
+constructor(@ForApplication val context: Context) : OnMapReadyCallback, LifecycleObserver {
 
     private lateinit var mapView: MapView
     private lateinit var map: GoogleMap
 
-    private val mapReadySubject: BehaviorSubject<Boolean> = BehaviorSubject.create()
+    open val shelterClickSubject = PublishSubject.create<LatLng>()
+    open val mapClickSubject = PublishSubject.create<LatLng>()
```

Comment:
Tried using higher order functions for this instead, but couldn't find a way to test it since this object is mocked using mockito in my VM test. Subjects work fine though.